src/graph.jl

```diff
@@ -82,7 +82,7 @@ visited: (optional, None) If given, a set of nodes that have
 
 NOTE: If node doesn't exist, the method will return False.
 """
-function ancestor_of(graph::ArbiterGraph, name::Symbol, ancestor::Symbol, visited::NodeSet)
+function ancestor_of(graph::ArbiterGraph, name::Symbol, ancestor::Symbol, visited::NodeSet=NodeSet())
```

Comment:
@omus I don't understand how this is reasonable. Seems like we just don't have tests which uncover the bug. NodeSet() is mutable and is mutated in the function. That will affect behaviour of future calls to this function.